client/components/table.jsx

```diff
@@ -5,40 +5,107 @@ class SummaryTable extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      hasChanged: false
+      modalOpen: false
     };
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  openModal() {
+    this.setState({
+      modalOpen: true
+    });
+  }
+
+  closeModal() {
+    this.setState({
+      modalOpen: false
+    });
   }
 
   render() {
     const calories = calorieCalculator(this.props.gender, this.props.age, this.props.height, this.props.goalWeight, this.props.activityLevel);
-    return (
-      <>
-      <h2 className="table-title">Today</h2>
-        <table>
-          <thead>
-            <tr>
-              <th colSpan="3">{calories} calories remaining</th>
-            </tr>
-          </thead>
-          <tbody>
-            <tr>
-              <td className="border-right">
-                0
+    if (this.state.modalOpen) {
+      return (
+        <>
+          <h2 className="table-title">Today</h2>
+          <table>
+            <thead>
+              <tr>
+                <th colSpan="3">{calories} calories remaining</th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr>
+                <td className="border-right">
+                  0
+              <p>consumed</p>
+                </td>
+                <td className="border-right">
+                  0
+              <p>burned</p>
+                </td>
+                <td>
+                  0
+              <p>net</p>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+          <div className="modal-button-container">
+            <button className="modal-button" onClick={this.openModal}>
+              <i className="fas fa-plus"></i>
+            </button>
+          </div>
+          <div className="modal-container" onClick={this.closeModal}>
+            <div className="modal">
+              <div>
+                <a href="#food-journal"><i className="fas fa-utensils"></i></a>
+                <p><a href="#food-journal" className="redirect-food-journal">Food journal</a></p>
+              </div>
+              <div>
+              <i className="fas fa-dumbbell"></i>
+              <p>Workout journal</p>
+              </div>
+            </div>
+          </div>
+        </>
```

Comment:
Instead of a big `if/else`, you can use `&&` conditional rendering in an expression

```jsx
{ this.state.modalOpen &&
  <>
    <div className="modal-button-container">
      <button className="modal-button" onClick={this.openModal}>
        <i className="fas fa-plus"></i>
      </button>
    </div>
    <div className="modal-container" onClick={this.closeModal}>
      <div className="modal">
        <div>
          <a href="#food-journal"><i className="fas fa-utensils"></i></a>
          <p><a href="#food-journal" className="redirect-food-journal">Food journal</a></p>
        </div>
        <div>
        <i className="fas fa-dumbbell"></i>
        <p>Workout journal</p>
        </div>
      </div>
    </div>
  </>
}
```

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Reply:
![Screen Shot 2021-04-12 at 2 36 39 PM](https://user-images.githubusercontent.com/74885918/114465851-885e0a80-9b9c-11eb-9d9e-c386346d5c50.png)